File: test-local-pkg/ts/tests/test-settings.ts
import { pathToFileURL } from 'url';
import { client, runTestSection, logResult } from "../utils/utils.js";

/**
 * TEST SETTINGS RESOURCE
 * Focuses on Message Templates, WhatsApp Templates, and User Management.
 */
async function testSettings() {
  console.log("\n" + "=".repeat(70));
  console.log("🚀 STARTING: SETTINGS & TEMPLATES TEST");
  console.log("=".repeat(70));

  const ts = Date.now();
  const state = {
    templateId: null as string | null,
  };

  try {
    // --- Section 1: Message Templates ---
    await runTestSection("Message Templates Lifecycle", async () => {
      // 1. Create Template
      const template = await client.settings.createMessageTemplate({
        name: `SDK Test Template ${ts}`,
        body: "Hello {{1}}, welcome to our service!",
        category: "UTILITY",
        language: "en_US"
      });
      state.templateId = template._id;
      logResult("Template Created", state.templateId);

      // 2. Get Template
      const fetched = await client.settings.getMessageTemplate(state.templateId!);
      logResult("Template Retrieved", fetched.name);

      // 3. List Templates
      const list = await client.settings.listMessageTemplates({ limit: 5 });
      logResult("Templates Count", list.data?.length);

      // 4. Update Template
      await client.settings.updateMessageTemplate(state.templateId!, {
        body: "Hi {{1}}, updated body!"
      });
      logResult("Template Updated", true);

      // 5. Search Templates
      const search = await client.settings.searchMessageTemplates({ q: `SDK Test` });
      logResult("Search Templates Count", search.length);
    }).catch(e => console.error(`\n❌ Section [Message Templates] Failed: ${e.message}`));

    // --- Section 2: WhatsApp Templates ---
    await runTestSection("WhatsApp Templates", async () => {
      const waTemplates = await client.settings.listWhatsAppTemplates({ limit: 5 });
      logResult("WhatsApp Templates (v1)", waTemplates.length);

      const waTemplatesV2 = await client.settings.listWhatsAppTemplatesV2({ limit: 5 });
      logResult("WhatsApp Templates (v2)", waTemplatesV2.length);
    }).catch(e => console.error(`\n❌ Section [WhatsApp Templates] Failed: ${e.message}`));

    // --- Section 3: User Management ---
    await runTestSection("User Management (Settings)", async () => {
      const users = await client.settings.listUsers({ limit: 5 });
      logResult("Users (Settings) Count", users.length);

      const rolesCount = await client.settings.getUserRolesCount();
      logResult("User Roles Distribution", rolesCount);
    }).catch(e => console.error(`\n❌ Section [User Management] Failed: ${e.message}`));

    console.log("\n" + "=".repeat(70));
    console.log("✅ SETTINGS TEST COMPLETED SUCCESSFULLY");
    console.log("=".repeat(70));

  } catch (err: any) {
    console.error("\n❌ FATAL TEST ERROR:", err.message);
    throw err;
  } finally {
    if (state.templateId) {
      console.log("\n[Cleanup] Removing message template...");
      await client.settings.deleteMessageTemplate(state.templateId!).catch(() => {});
    }
  }
}

if (import.meta.url === pathToFileURL(process.argv[1]).href) {
  testSettings().catch(console.error);
}

export { testSettings };

File: test-local-pkg/py/tests/test_settings.py
import time
from utils.utils import client, run_test_section, log_result

def test_settings():
    print("\n🚀 Testing Settings Resource...")

    state = {"template_id": None}

    # 1. Message Templates
    def templates_ops():
        res = client.settings.list_message_templates()
        log_result("Templates Count", len(res.get("data", [])))

        created = client.settings.create_message_template({
            "name": f"SDK_PY_TEMP_{int(time.time())}",
            "body": "Hello {{1}}",
            "category": "UTILITY",
            "language": "en_US"
        })
        state["template_id"] = created.get("_id")
        log_result("Template Created", state["template_id"])

        if state["template_id"]:
            fetched = client.settings.get_message_template(state["template_id"])
            log_result("Template Retrieved", fetched.get("name"))

            client.settings.update_message_template(state["template_id"], {"body": "Hi {{1}}, updated!"})
            log_result("Template Updated", True)

            search = client.settings.search_message_templates(q="SDK_PY")
            log_result("Search Templates Count", len(search))

            client.settings.delete_message_template(state["template_id"])
            log_result("Template Deleted", True)
    run_test_section("Message Templates Ops", templates_ops)

    # 2. WhatsApp Templates
    def wa_templates():
        wa = client.settings.list_whatsapp_templates(limit=5)
        log_result("WA Templates (v1)", len(wa))

        try:
            wa_v2 = client.settings.list_whatsapp_templates_v2(limit=5)
            log_result("WA Templates (v2)", len(wa_v2))
        except:
            print("   [Skip] WA Templates v2 failed")
    run_test_section("WhatsApp Templates", wa_templates)

    # 3. User Management
    def user_mgmt():
        users = client.settings.list_users(limit=5)
        log_result("Users Count", len(users))

        try:
            roles = client.settings.get_user_roles_count()
            log_result("User Roles Count", roles)
        except:
            print("   [Skip] get_user_roles_count failed")
    run_test_section("User Management (Settings)", user_mgmt)

    print("\n✅ Settings Resource Testing Completed.")

if __name__ == "__main__":
    test_settings()
